### PR TITLE
🐛 ignore Conflict status in order to make instance able to detach interface

### LINKED ISSUE #959: VM instances with error state cann't be deleted successfully
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
1. Create a workload cluster.
2. Failed to create worker intances because of some exception of nova, state of instances are "error".
3. Delete this workload cluster.
4. Worker instances are not deleted successfully.

**What did you expect to happen:**
Worker instances can be deleted successfully.

**Anything else you would like to add:**
We should check state of instance before detaching interfaces from instance, because it is not allowed to detach interfaces from an error instance.

**Environment:**

- Cluster API Provider OpenStack version (Or `git rev-parse HEAD` if manually built): v0.4.0
- Cluster-API version: v0.4.0
- OpenStack version: Rocky
- Minikube/KIND version:
- Kubernetes version (use `kubectl version`): 1.20.10
- OS (e.g. from `/etc/os-release`): centos-8

### PR DESCRIPTION
**Pull request by jichenjc**
…face

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #959 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -61,6 +61,19 @@ func (mc *MetricPrometheusContext) ObserveRequestIgnoreNotFound(err error) error
 	return mc.ObserveRequest(err)
 }
 
+// ObserveRequestIgnoreNotFoundorConflict records the request latency and counts the errors if it's not IsNotFound or IsConflict.
+func (mc *MetricPrometheusContext) ObserveRequestIgnoreNotFoundorConflict(err error) error {
```
Comment (tobiasgiese) [inline pkg/metrics/metrics.go:65]:
```suggestion
func (mc *MetricPrometheusContext) ObserveRequestIgnoreNotFoundOrConflict(err error) error {
```

**Review by tobiasgiese — COMMENTED** [review]
Found one tiny nit

**Comment by jichenjc** [comment]
/hold cancel

the tiny int can be updated later on :)

**Pull request merged**